### ISSUE TITLE
[BITV] 9.1.4.1/3.1 - Styling menubar - active contrast

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/text#3860]
For many elements like "B" and "I", selected styles are visually marked by a pale blue background (1.5:1 to white) and a change of text color (black to blue, 2.2:1). Both remain below the contrast difference of 3:1 that is needed to indicate the state visually. (2)

<img width="237" alt="7d269fef5d1ea6ddaaa3bbb3e5e4230b" src="https://user-images.githubusercontent.com/1315170/222219980-b1ea87af-e3be-4fb5-b9a8-3b590d5aea06.png">

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-45ae33ca9f-v3-n1

</details>

- [ ] 9.1.4.1/10.1 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-45ae33ca9f-v10-n1

**Comment by juliushaertl** [comment]
@jancborchardt I remember we had discussed the current style, but to match the constrast we'd need to use rather black as the color for the active indicator. 

Shall we just switch to the secondary style as the active indicator or would you have any other suggestion?

For comparison the NcButton component, but they don't have dedicated active states

![Image](https://user-images.githubusercontent.com/3404133/222267361-fbe26b4c-194d-4256-8e1a-0c4fc1a69764.png)

**Comment by AndyScherzinger** [comment]
@jancborchardt what do you think about https://github.com/nextcloud/text/issues/3860#issuecomment-1450862865 -  pending decision is a blocker for resolving the issue by the devs.

**Comment by jancborchardt** [comment]
> Shall we just switch to the secondary style as the active indicator or would you have any other suggestion?

Yes, that sounds the most reasonable and consistent.

**Comment by jancborchardt** [comment]
@AndyScherzinger Why does it not work? The contrast as used for the secondary button is even triple-A compliant:
![image](https://github.com/nextcloud/text/assets/925062/11c53894-45fe-4020-9a97-072a2260e489)

**Comment by JuliaKirschenheuter** [comment]
> @AndyScherzinger Why does it not work? The contrast as used for the secondary button is even triple-A compliant: ![image](https://user-images.githubusercontent.com/925062/239899585-11c53894-45fe-4020-9a97-072a2260e489.png)

`color` is absolutely ok. But the `background-color` to the white background is not ok ;(
How could we move with it?

And which possibility do we have for the dark mode?

**Comment by jancborchardt** [comment]
Then we either need to use the primary style for highlighting (which might be a bit much), or we do it again like we had some time before with the small dot below the active elements. @juliushaertl do we still have CSS for that by any chance?

And if you know what I mean, do you have a preference? I think the version with the dot below the active icon probably looks a bit cleaner.

**Comment by juliushaertl** [comment]
I would prefer the primary then, as the dot is not very common among other editing tools and I think we removed it in the past because it was not easy to discover and looked more like a notification.

I've also looked at other tools and noticed that except apple, all seem to stick to a light background with doesn't match contrast requirements. 

Could we come up with a global definition of how active style of a button or element is indicated, as the current style is also used in other places or there are other parts where the contrast doesn't match in a similar way:

- https://github.com/nextcloud/nextcloud-vue/pull/3945
- App navigation active style (e.g. talk/mail)
- App content list (e.g. mail messages)

**Comment by jancborchardt** [comment]
@juliushaertl you're right, for active app nav and app content list we already are switching to primary style as per @JuliaKirschenheuter, then we should indeed do it here too, and for any other active style.

**Comment by JuliaKirschenheuter** [comment]
@jancborchardt ok, does it means that all active elements like

<img width="237" alt="7d269fef5d1ea6ddaaa3bbb3e5e4230b" src="https://user-images.githubusercontent.com/1315170/222219980-b1ea87af-e3be-4fb5-b9a8-3b590d5aea06.png">

would have styles from primary (button), like

![Image](https://user-images.githubusercontent.com/3404133/222267361-fbe26b4c-194d-4256-8e1a-0c4fc1a69764.png)

right? 


Another alternative would be to use high contrast border. What would you prefer?

**Comment by jancborchardt** [comment]
@JuliaKirschenheuter yes, we can hse primary as indicator for active items. This is what e.g. iOS Notes does as well, as @juliushaertl mentioned.

---

Additionally but separately, we need to fix our secondary button style as it's not accessible.

If we look at Material Design as reference, they have:
https://m3.material.io/components/all-buttons
- 2 Filled button: our primary
- 3 Filled tonal button: our secondary – so theirs is not accessible?
- 4 Outlined button: **We could move our secondary to this?**
- 5/6 Text/icon button: Our tertiary

@nextcloud/designers what do you think? I'd say just moving to outline style for secondary button is simplest and safest.

**Comment by JuliaKirschenheuter** [comment]
by the way, Material Design `is not immediately` accessible. 
Here is a same problem 
![Screenshot from 2023-05-31 15-37-21](https://github.com/nextcloud/text/assets/6078378/e16aad72-8eb7-4d28-9231-59f53aed5a85)
as was reported in our issue: https://github.com/nextcloud/server/issues/36947

**Comment by jancborchardt** [comment]
Yep exactly. So we should move our secondary button from tonal to outlined, so that when we use it, it is accessible. :)

**Comment by juliushaertl** [comment]
As mentioned in chat outline is not very clear to differentiate then to the focus state. 

@jancborchardt I think we need the design team here to jump in and bring up mockups as clear definition for the following buttons adding additional states.

Might be enough to define those for tertiary buttons for the use cases I know of.

## States we have

### Normal
![Screenshot 2023-06-14 at 10 47 36](https://github.com/nextcloud/text/assets/3404133/387300a2-fb50-41e0-bb41-ca5ab52acde9)


### Disabled
![Screenshot 2023-06-14 at 10 47 41](https://github.com/nextcloud/text/assets/3404133/28688331-ea66-4aa9-a7ac-9b4246c062d2)

### Focus
![Screenshot 2023-06-14 at 10 49 07](https://github.com/nextcloud/text/assets/3404133/93a6a903-8bff-4a0d-98e6-473390556990)

## States we'd need definition

- [ ] Active
- [ ] Active focus

## References of existing usages of this case

### Text

![Screenshot 2023-06-14 at 10 54 31](https://github.com/nextcloud/text/assets/3404133/8fefecc0-c7a8-4e0c-8098-07add2b0103e)

### Collabora

![Screenshot 2023-06-14 at 10 56 04](https://github.com/nextcloud/text/assets/3404133/7cceeb91-bc92-43ae-a5f0-2dc242ca59f9)

## AppNavigationItem (not a button but should probably use a similar style)

![Screenshot 2023-06-14 at 10 56 43](https://github.com/nextcloud/text/assets/3404133/52cdb546-43ff-4c97-b9be-b3d8f92fed6c)

**Comment by JuliaKirschenheuter** [comment]
Suggestion for hover (like focus state)

### Hover 

![Screenshot 2023-06-14 at 10 49 07](https://github.com/nextcloud/text/assets/3404133/93a6a903-8bff-4a0d-98e6-473390556990)

**Comment by JuliaKirschenheuter** [comment]
https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

**Comment by luka-nextcloud** [comment]
Do we have final decision and a summary of it here?

**Comment by juliushaertl** [comment]
@JuliaKirschenheuter We would still need to switch to a different active style, right? We can take care of that using https://github.com/nextcloud/nextcloud-vue/pull/4344, but maybe you can quickly confirm that this is required.

**Comment by susnux** [comment]
@JuliaKirschenheuter is this fixed? I thought only the focus issues was solved, what about the active state?
